src/plugins/kibana_utils/common/persistable_state/index.ts

```diff
@@ -47,8 +52,29 @@ export interface PersistableState<P extends SerializableState = SerializableStat
    * @param state
    */
   extract: (state: P) => { state: P; references: SavedObjectReference[] };
+
+  /**
+   * migrateToLatest function receives state of older version and should migrate to the latest version
+   * @param state
+   * @param version
+   */
+  migrateToLatest?: (state: SerializableState, version: string) => P;
```

Comment:
I'm a little confused about why `migrateToLatest` requires a `version` at all? Shouldn't that just be something that the function knows about internally? Or maybe `version` is used differently here from `migrate` and that's what I am getting mixed up about

Reply:
yeah, this is the version of your state, where in migrate() its the version of migration you want to run. i will make that clear in the jsdocs

Reply:
Ah okay - yeah, I think the fact that `version` means two things is just a naming difficulty then

Reply:
The doc comments clear up most of the confusion, though I think something like toVersion and fromVersion would be even more intuitive.